pinot-query-runtime/src/main/java/org/apache/pinot/query/runtime/operator/AggregateOperator.java

```diff
@@ -325,57 +291,126 @@ private ExpressionContext convertRexExpressionToExpressionContext(int aggIdx, in
     return exprContext;
   }
 
-  // TODO: If the previous block is not mailbox received, this method is not efficient.  Then getDataBlock() will
-  //  convert the unserialized format to serialized format of BaseDataBlock. Then it will convert it back to column
-  //  value primitive type.
-  static Map<ExpressionContext, BlockValSet> getBlockValSetMap(AggregationFunction aggFunction, TransferableBlock block,
-      DataSchema inputDataSchema, Map<String, Integer> colNameToIndexMap, int filterArgIdx) {
+  private int[] getGroupKeyIds(List<RexExpression> groupSet) {
```

Comment:
nit: why reordering functions? this is original getGroupSet right? (and if not cant we remove the `convertRexExpressionToExpressionContext` func)?

Reply:
It is quite different though. One retrieves the expressions, one just for the ids. I'll try to remove `convertRexExpressionToExpressionContext` in a separate PR because that involves quite some change on the aggregation function